### PR TITLE
HZN-352: Document Newts setup

### PR DESCRIPTION
**Pull request by indigo423**
Initialized setup guide based on OpenNMS Wiki and GitHub docs. Added description how to install Cassandra on RHEL and Debian-based systems. Described how to install Newts keyspace and verify the installation.

JIRA: http://issues.opennms.org/browse/HZN-352
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS103-13
Deployed: http://docs.opennms.org/opennms/branches/jira-HZN-352/guide-install/guide-install.html

### CODE REVIEW

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gi-install-cassandra-debian]]
+==== Installing on Debian-based systems
+
+This section describes how to install the latest _Cassandra 2.1.x_ release on a _Debian_ based systems for _Newts_.
+The first steps add the _DataStax_ community repository and install the required _GPG Key_ to verify the integrity of the _DEB packages_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc]: The first step is to add... . After that install the...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gi-install-cassandra-debian]]
+==== Installing on Debian-based systems
+
+This section describes how to install the latest _Cassandra 2.1.x_ release on a _Debian_ based systems for _Newts_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc]: Debian based or Debian-based

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+The first steps add the _DataStax_ community repository and install the required _GPG Key_ to verify the integrity of the _DEB packages_.
+Installation of the package is done with _apt_ and the _Cassandra_ service is added to the run level configuration.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc]: run level or runlevel or run-level

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+NOTE: This description is build on _Debian 8_ and _Ubuntu 14.04 LTS_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc]: ...was built...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+.Add DataStax repository
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc]: Add the _DataStax_ repository

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+----
+
+.Install latest Cassandra 2.1.x package
+[source, bash]
+----
+apt-get update
+apt-get install dsc21=2.1.8-1 cassandra=2.1.8
+----
+
+The _Cassandra_ service is added to the run level configuration and is automatically started after installing the package.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc]: run level or runlevel or run-level

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+TIP: Verify if the _Cassandra_ service is automatically started after rebooting the server.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-debian.adoc]: Verify whether the...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc [hunk 8]
```diff
@@ -0,0 +1,54 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gi-install-cassandra-rhel]]
+==== Installing on RHEL-based systems
+
+This section describes how to install the latest _Cassandra 2.1.x_ release on a _RHEL_ based systems for _Newts_.
+The first steps add the _DataStax_ community repository and install the required _GPG Key_ to verify the integrity of the _RPM packages_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc]: The first step is to add... . After that install the...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+Installation of the package is done with _yum_ and the _Cassandra_ service is managed by _Systemd_.
+
+NOTE: This description is build on _RHEL 7_ and _CentOS 7.1_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc]: ...was built...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+.Add DataStax repository
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc]: Add the _DataStax_ repository

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+chkconfig cassandra on
+----
+
+.Start cassandra service
+[source, bash]
+----
+service cassandra start
+----
+
+TIP: Verify if the _Cassandra_ service is automatically started after rebooting the server.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-rhel.adoc]: Verify whether the...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc [hunk 12]
```diff
@@ -0,0 +1,26 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gi-install-cassandra-windows]]
+==== Installing on Windows Server systems
+
+This section describes how to install the latest _Cassandra 2.1.x_ release on a _Microsoft Windows Server_ based systems for _Newts_.
+The first steps download the graphical installer and register _Cassandra_ as a _Windows Service_ and can be manged through the _Service Manager_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc]: First, download the...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+NOTE: This description is build on _Windows Server 2012_.
+
+.Download the DataStax graphical installer for Cassandra from PowerShell or a Browser
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc]: _DataStax_

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gi-install-cassandra-windows]]
+==== Installing on Windows Server systems
+
+This section describes how to install the latest _Cassandra 2.1.x_ release on a _Microsoft Windows Server_ based systems for _Newts_.
+The first steps download the graphical installer and register _Cassandra_ as a _Windows Service_ and can be manged through the _Service Manager_.
+
+NOTE: This description is build on _Windows Server 2012_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc]: ...was built...

File: opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+Invoke-WebRequest http://downloads.datastax.com/community/datastax-community-64bit_2.1.8.msi -Outfile datastax-community-64bit_2.1.8.msi
+----
+
+Run the Windows Installer file from _PowerShell_ or through _Windows Explorer_ and follow the setup wizard to install.
+During the installation, accept the options to automatically start the services.
+By default the _DataStax Server_, _OpsCenter Server_ and the _OpsCenter Agent_ will be automatically installed and started.
+
+NOTE: The _DataStax OpsCenter Server_ is only required to be installed once per _Cassandra Cluster_.
+
+IMPORTANT: If you install the _DataStax OpsCenter_ make sure you have _Chrome_ or _Firefox_ as browser installed.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/cassandra-windows.adoc]: If you install the _DataStax OpsCenter_ make sure you have _Chrome_ or _Firefox_ installed.

File: opennms-doc/guide-install/src/asciidoc/text/newts/configure-opennms.adoc [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+:imagesdir: ../../images
+
+[[gi-configure-opennms]]
+=== Configure OpenNMS
+
+Once you have _OpenNMS_, _Cassandra_ and the _Newts_ keyspace installed and up and running, the _OpenNMS_ application has to be configured.
+To be able to use _Newts_ the graphing library has to be changed from _RRDTool_ or _JRobin_ to _Backshift_.
+The following properties for _OpenNMS_ has to be changed in `/etc/opennms/opennms.properties`:
+
+.Configuration properties change for OpenNMS in opennms.properties
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/configure-opennms.adoc]: _OpenNMS_

File: opennms-doc/guide-install/src/asciidoc/text/newts/introduction.adoc [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+:imagesdir: ../../images
+
+[[gi-install-ts-newts]]
+== Installing Time Series database Newts
+
+link:http://newts.io/[Newts] is a time-series data store based on link:http://cassandra.apache.org/[Apache Cassandra].
+_Newts_ is a persistence strategy, that can be used as an alternative to link:http://www.opennms.org/wiki/JRobin[JRobin] or link:http://oss.oetiker.ch/rrdtool/[RRDtool].
+
+IMPORTANT: It is currently not supported to initialize the _Newts_ keyspace from _Microsoft Windows Server_ operating system.
+           _Microsoft Windows_ based _Cassandra_ server can be part of the cluster, the initializing can just be executed from a _Linux_ system.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/introduction.adoc]: ...cluster, but keyspace initialization is only possible using a _Linux_-based system.

File: opennms-doc/guide-install/src/asciidoc/text/newts/introduction.adoc [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+
+[[gi-setup-cassandra]]
+=== Setting up Cassandra
+
+It is recommended to install _Cassandra_ on a dedicated server, but is also possible to run a node on the OpenNMS server itself.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/introduction.adoc]: _OpenNMS_

File: opennms-doc/guide-install/src/asciidoc/text/newts/introduction.adoc [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+This installation guide describes how to setting up a single _Cassandra_ for evaluating and testing _Newts_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/introduction.adoc]: ...how to set up a ...

File: opennms-doc/guide-install/src/asciidoc/text/newts/setup-newts.adoc [hunk 20]
```diff
@@ -0,0 +1,48 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gi-setup-newts]]
+=== Setting up Newts
+
+The setup for _Newts_ has two parts, the first one initializes the keyspace in the _Cassandra_ storage and the second part is the standalone running _Newts API_.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/setup-newts.adoc]: ...has two parts: The first...

File: opennms-doc/guide-install/src/asciidoc/text/newts/setup-newts.adoc [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+Both parts are packaged and maintained in the link:https://github.com/OpenNMS/newts/releases[Newts GitHub repository].
+The latest pre-compiled tarball can be downloaded as link:https://github.com/OpenNMS/newts/releases/download/1.3.0/newts-1.3.0-bin.tar.gz[newts-1.3.0-bin.tar.gz]
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/setup-newts.adoc]: We need to change this for each newts-release? Isn't there something like newts-LATEST-bin.tar.gz?

File: opennms-doc/guide-install/src/asciidoc/text/newts/setup-newts.adoc [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+.Download and extract pre-compiled tarball for Newts on Linux
+[source, bash]
+----
+curl -L https://github.com/OpenNMS/newts/releases/download/1.3.0/newts-1.3.0-bin.tar.gz \
+     -o newts-1.3.0-bin.tar.gz
+tar xvf newts-1.3.0-bin.tar.gz
+----
+
+The default _Newts_ config (`etc/config.yaml`) will connect to _localhost_ at the default port and keyspace, but if you're using your own install of _Cassandra_ then you'll need to edit it accordingly.
+To perform a one-time initialization of _Newts_, and create the _Cassandra_ schema.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/newts/setup-newts.adoc]: For one-time initialization of _Newts_ and the creation of the _Cassandra_ schema perform the following steps:

**Comment by indigo423** [comment]
Thank you for the review :+1:

**Comment by agalue** [comment]
If I'm not mistaken, the NewTS integration requires storeByGroup. I think the documentation must be updated to tell the users to enable storeByGroup (as I changed on the old wiki page).

**Pull request merged**